### ISSUE TITLE
[hackerone] Tor DNS issue

### ISSUE THREAD

**Issue by diracdeltas** [brave/brave-browser#13527]
https://hackerone.com/reports/1077022

related: #4257. it appears this was a known issue at least in the muon days. unclear if it was ever fixed in brave-core.

https://sites.google.com/a/chromium.org/dev/developers/design-documents/network-stack/socks-proxy suggests that DNS prefetching may be the only source of this leak, and supposedly this can be disabled via the 'preload pages' setting in brave://settings/cookies.

**Comment by stephendonner** [comment]
Verified `FIXED` on `1.20.108` using the testplan in https://github.com/brave/brave-core/pull/7769, on:

```
Brave | 1.20.108 Chromium: 88.0.4324.182 (Official Build) (x86_64)
Revision | 73ee5087001dcef33047c4ed650471b225dd8caf-refs/branch-heads/4324@{#2202}
OS
macOS Version 11.2.1 (Build 20D74)
```

```
Brave | 1.20.108 Chromium: 88.0.4324.182 (Official Build) (64-bit)
Revision73ee5087001dcef33047c4ed650471b225dd8caf-refs/branch-heads/4324@{#2202}
OS | Windows 10 OS Version 2009 (Build 21318.1000)
```

Verified that in both the `DoH off` and `DoH enabled` cases, I didn't see `tools.ietf.org` via the DNS filter in Wireshark; due to time pressure + priorities, will update this with screenshots and annotations, after hot-fix release testing 👍

**Comment by peebowo** [comment]
bruh
